Testsuite fails on ppc64le

On the POWER8 `ppc64le` system
```
Linux sardonis 3.19.0-15-generic #15-Ubuntu SMP Thu Apr 16 23:32:13 UTC 2015 ppc64le ppc64le ppc64le GNU/Linux
```

I get the following reproducible test suite failure with `arb` 2.7.0:
```
make[1]: Entering directory '/home/jdemeyer/sage/local/var/tmp/sage/build/arb-2.7.0.p0/src/arf'
[...]
abs_bound_le_2exp_fmpz....PASS
complex_sqr....FAIL!
prec = 698, rnd = 0

a = (6339285465397756945139231368132496554128019744394747208275350942065595714223327282882004073897576134671584866440587073589025503475990026413326511053883607915299395946163061310275939887721284702577049600905500587493134063493826068090023461124398687901692686283320433763252148270486039504843224058218698609943517081716892849130905696941657348763839640087212158083454683180059633412954189769703715978964001161003786946680425738899972518273473719571986450603741721249271434462214845897250532971870021425902922261780901847636017986587369648930793121015247658270597452971859383674118378417538423342960709853716020960438322933655567693176273354358777068473650766777407727302868991 * 2^-2235)

b = (20769147822475905021182172033122559 * 2^-113)

e1 = (1315013909709381559722549835740549443786829800454131997155171828694211243086620368814692247495737405770735789621415903963936315776768232380505812958907618889969868245794264502147656964862275963676046558834458623 * 2^-698)

f1 = (1315031467311928423708944301303640969760880181222735669239969305516422962189234000124364732949570062519188408040675832296657303229505388769788357929947309747087029353794143201921833529433945488716348391784710143 * 2^-696)

e2 = (-323517714603253067973887569535021297144815213232427194808985473711617 * 2^-226)

f2 = (1315031467311928423708944301303640969760880181222735669239969305516422962189234000124364732949570062519188408040675832296657303229505388769788357929947309747087029353794143201921833529433945488716348391784710143 * 2^-696)

r1 = 3, r2 = 3
../Makefile.subdirs:84: recipe for target '../build/arf/test/t-complex_sqr_RUN' failed
make[1]: *** [../build/arf/test/t-complex_sqr_RUN] Aborted
make[1]: Leaving directory '/home/jdemeyer/sage/local/var/tmp/sage/build/arb-2.7.0.p0/src/arf'
```

The compiler is vanilla GCC 4.9.3. I compiled `arb` and its dependencies (`mpir`, `mpfr`, `flint`) with `-O0` to rule out compiler errors and it did not make a difference.

Running Sage with this version of `arb` also gives errors in the Sage test suite in functions `arb_digamma`, `arb_const_catalan` and `arb_polylog_si`.

```
submul....FAIL (aliasing 1)!
prec = 1205, rnd = 0

x = (-4835629924954783359180799 * 2^590296089634659107757)

z = (-550921269894147237274648685022492079096825010962694736744659493673166255653731244523903087166053597074151219798505686543542860875901512977615388900987381708170637
701163915919785751946932628366383699977739349162989141690998161200993630275790250218486500863280092258474016081283403150008418887162950055787067856968752537042818076514
986762887003147435869307309916159 * 2^1180592179269318214386)

v = (-550974626591539860560763282829084129475332552717879396323243043047417420810007918891195843618781505557905406663902581015378675555766783033455756235274137921681885568956354969059049261193543916656642911489558399900981278586113663803340481340601109759356649169932184710072142932319852375708782659897301898677304206931833033099900776697376907167605152015328426852351 * 2^1180592179269318214473)

r1 = 1, r2 = 1
../Makefile.subdirs:84: recipe for target '../build/arf/test/t-submul_RUN' failed
```

--- a/arf.h
+++ b/arf.h
@@ -870,12 +870,12 @@ void arf_randtest_special(arf_t x, flint_rand_t state, slong bits, slong mag_bit
         } \
         else if ((_xn) == 2) \
         { \
-            mp_limb_t __x1, __x0, __y1, __y0; \
-            __x0 = (_x)[0]; \
-            __x1 = (_x)[1]; \
-            __y0 = (_y)[0]; \
-            __y1 = (_y)[1]; \
-            nn_mul_2x2((_z)[3], (_z)[2], (_z)[1], (_z)[0], __x1, __x0, __y1, __y0); \
+            mp_limb_t __arb_x1, __arb_x0, __arb_y1, __arb_y0; \
+            __arb_x0 = (_x)[0]; \
+            __arb_x1 = (_x)[1]; \
+            __arb_y0 = (_y)[0]; \
+            __arb_y1 = (_y)[1]; \
+            nn_mul_2x2((_z)[3], (_z)[2], (_z)[1], (_z)[0], __arb_x1, __arb_x0, __arb_y1, __arb_y0); \
         } \
         else if ((_x) == (_y)) \
         { \

